client/hooks/packs/usePackTable.tsx

```diff
@@ -0,0 +1,127 @@
+import { useDispatch } from "react-redux";
+import { useSelector } from "../redux/useSelector";
+import { useState } from "react";
+import { duplicatePackItem } from "~/store/packsStore";
+import { ItemCategoryEnum } from "~/constants/itemCategory";
+import { convertWeight } from "~/utils/convertWeight";
+ 
+
+export const usePackTable = ({
```

Comment:
this might be a good spot to use our new useTable hook when its ready. That will help with some of this logic